src/port/port.tcl

```diff
@@ -3927,6 +3927,128 @@ proc action_echo { action portlist opts } {
     return 0
 }
 
+proc prompt { message default variable } {
+    puts -nonewline $message
+    flush stdout
+    gets stdin variable
+    if {[string trimleft $variable] eq ""} {
+        set variable $default
+    }
+    return 0
+}
+
+proc offer_clone {} {
+    global macports::os_platform macports::os_major macports::os_arch \
+        macports::sources macports::autoconf::rsync_path \
+        macports:rsync_options macports::getindex macports::sources_default
+
+    set msg4 {Rsyncing a PortIndex file...}
+    set msg5 {}
+
+    # OFFER GIT-BASED SOURCE AND ASK FOR PATH
+    set msg1 "You do not seem have a working copy of the macports-ports git repository, which is required to contribute this change."
+    set retvalue [$macports::ui_options(questions_yesno) $msg1 "clone" {} {y} 0 "Create one now?"]
+    if {$retvalue == 1} {
+        # quit as user answered 'no'
+        return 0
+    }
+
+    set path [$macports::ui_options(questions_input) "Where do you want to create your working copy?" [file join [file normalize ~] macports-ports] 0]
+
+    # WALK DOWN FROM PATH TO ROOT AND ASK FOR PERMISSIONS IF NEEDED
+    set paths {}
+    set temp $path
+    while {$temp != [file normalize /]} {
+        ui_msg "$temp [file readable $temp]"
+        if {[file readable $temp]} {
+            break
+        }
+        lappend paths $temp
+        set temp [file dirname $temp]
+    }
+    set msg2 "The 'macports' user needs search access to ~/macports-ports to use ports in
+this directory. The following directories will be changed to include 'x'
+permissions for group and others:
+"
+    if {[llength $paths] > 0} {
+        set allow [$macports::ui_options(questions_yesno) $msg2 "allow" $paths {y} 0 "Do you want to allow this?"]
+        if {$allow == 1} {
+            return 0
+        }
+    }
+
+    # CREATE GIT SOURCE BY CLONING MACPORTS-PORTS
+    if {[catch {system "${macports::findBinary git} clone --depth 100 $url $path"} result]} {
```

Comment:
This should be `[macports::findBinary git]`. `${macports::findBinary git}` would look for the variable named `findBinary git` in the `macports` namespace. No idea whether Tcl variables can contain spaces – wouldn't be surprised if it's possible.